MediaBrowser.Providers/TV/TheTVDB/TvDbClientManager.cs

```diff
@@ -218,5 +232,34 @@ private async Task<T> TryGetValue<T>(object key, Func<Task<T>> resultFactory)
                 _cacheWriteLock.Release();
             }
         }
+
+        private static string GenerateKey(params object[] objects)
+        {
+            var key = string.Empty;
+
+            foreach (var obj in objects)
+            {
+                var objType = obj.GetType();
+                if (objType.IsPrimitive || objType == typeof(string))
+                {
+                    key += obj + ";";
+                }
+                else
+                {
+                    foreach (PropertyInfo propertyInfo in objType.GetProperties())
+                    {
+                        var currentValue = propertyInfo.GetValue(obj, null);
+                        if (currentValue == null)
+                        {
+                            continue;
+                        }
+
+                        key += propertyInfo.Name + currentValue + ";";
```

Comment:
should probably be
```cs
key += propertyInfo.Name + "=" + currentValue + ";";
```
to avoid potential funny things in values in some properties coinciding with names in others